rln/src/public.rs

```diff
@@ -550,6 +553,64 @@ mod test {
         assert_eq!(root_delete, root_empty);
     }
 
+    #[test]
+    // We test leaf setting with a custom index, to enable batch updates to the root
+    // Uses `set_leaves_from` to set leaves in a batch, from index `start_index`
+    fn test_leaf_setting_with_index() {
+        let tree_height = TEST_TREE_HEIGHT;
+        let no_of_leaves = 256;
+        let start_index = 5;
```

Comment:
I don't think it should be random, since unit tests should be deterministic. Maybe we can have multiple runs of the same test with start_index = 0..no_of_leaves, but it seems excessive. wdyt?

Reply:
Addressed in [ed6084a](https://github.com/vacp2p/zerokit/pull/63/commits/ed6084a4d175dca37700a78342cace49e8423689)